vnext/Microsoft.ReactNative/Views/TextViewManager.cpp

```diff
@@ -43,18 +46,17 @@ class TextShadowNode final : public ShadowNodeBase {
   }
 
   void AddView(ShadowNode &child, int64_t index) override {
+    auto &childNode = static_cast<ShadowNodeBase &>(child);
+    VirtualTextShadowNode::ApplyTextTransformToNode(childNode, textTransform, /* hasChanged = */false);
+
+    auto run = childNode.GetView().try_as<winrt::Run>();
     if (index == 0) {
-      auto run = static_cast<ShadowNodeBase &>(child).GetView().try_as<winrt::Run>();
       if (run != nullptr) {
         m_firstChildNode = &child;
         auto textBlock = this->GetView().as<xaml::Controls::TextBlock>();
```

Comment:
we don't currently have RichTextBlock inbox, but if we did, this would need to be rewritten to allow RichTextBlock too

Reply:
RichTextBlock would likely be a different root text component than TextBlock. But yes, if we change from TextBlock to RichTextBlock, this will need some adjustment (as I believe we'll need a top-level Paragraph inline in the RichTextBlock).

Reply:
It would probably be better to create a base shadow node type with a field that discriminates between virtual text (Span) and raw text (Run), but given that React Native's ABI-safe IViewManager implementation only supports FrameworkElement and higher right now (see #5539), a react-native-windows user would not be able to create a ViewManager for Inlines that would break this today.

Reply:
react-native-xaml creates Inlines/Runs and other non-frameworkelements
when a type is not an FE, I create a ContentControl and set the Content to be the non-FE element.
Then in AddView, I unwrap it from the ContentControl and set the right parent/child relationship (as well as transfer the Tag id)

Reply:
plus one thing to note that might be relevant: we have mixed-mode apps where part of the content is being created directly in XAML and other content in RNW, so it's important to make sure there are no assumptions made about how we might treat Run/Span/etc. differently inside of RNW (e.g. if we set an attached property on Runs created inside RNW, we should not expect all Runs to have that tag since some can come from outside)

Reply:
@asklar - The react-native-windows TextViewManager does not do any ContentControl unwrapping today, that's an implementation detail of your plugin. We can safely assume that any Run or Span that is created *today* that can be added as a child to the TextBlock or a virtual text Span will come from either VirtualTextViewManager or RawTextViewManager.

If you'd prefer, I'm happy to change this "switch" behavior into a base shadow node class for inlines as an enum, but it's solving a problem that will not exist until a significant change to react-native-windows occurs (FrameworkElement -> DependencyObject base type for IViewManager). And the point at which that happens, we'll probably be on to Fabric and all these ViewManagers will be deprecated :D

Reply:
Actually, I don't think this should change right now. I can't take the base shadow node class approach because I'm just shifting the assumption in an even less safe way to a static cast on the ShadowNode reference. At least the current approach uses try_as.

I suggest we keep the check as is for now, and at some point in the future (that may never come), if someone needs to extend this behavior to support user supplied or mixed-mode supplied Inlines, then we can fix then.

Reply:
Changed my mind - we can use `node->GetViewManager()->GetName()` to check what type of view we're dealing with.

Reply:
@rozele yes, it's an implementation detail of react-native-xaml but my point was more about making you aware that we have these hybrid apps, and where xaml objects can come into RNW even when they're not framework elements.
 FYI there is already new API for view managers that operates on `IInspectable`: [IViewManagerCreateWithProperties](https://microsoft.github.io/react-native-windows/docs/next/IViewManagerCreateWithProperties).

We still expect to get a FrameworkElement internally, but I'm hoping we can relax this in the future, and the fact that the API is based on IInspectable allows to do this transparently for the modules/apps out there that may use this new API.